modules/cudf/src/data_frame.ts

```diff
@@ -253,13 +264,75 @@ export class DataFrame<T extends TypeMap = any> {
    */
   filter(mask: Series<Bool8>) {
     const temp       = new Table({columns: this._accessor.columns});
-    const columns    = temp.gather(mask._col);
+    const columns    = temp.gather(mask._col, false);
     const series_map = {} as SeriesMap<T>;
     this._accessor.names.forEach(
       (name, index) => { series_map[name] = Series.new(columns.getColumnByIndex(index)); });
     return new DataFrame(series_map);
   }
 
+  /**
+   * Join columns with other DataFrame.
+   *
+   * @param props the configuration for the join
+   * @returns the joined DataFrame
+   */
+  join<U extends TypeMap, R extends keyof U>(props: JoinProps<U, R>): DataFrame<any> {
+    const {how, other, on} = props;
+    const nullEquality     = props.nullEquality == undefined ? true : props.nullEquality;
+    const lsuffix          = props.lsuffix == undefined ? '' : props.lsuffix;
+    const rsuffix          = props.rsuffix == undefined ? '' : props.rsuffix;
```

Comment:
What about this instead?
```suggestion
    const { how, other, on, nullEquality = true, lsuffix = '', rsuffix = '' } = props;
```

Reply:
I keep forgetting this is a thing 😂